File: VB/ThreadedSkinning/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.Threading
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraEditors

Namespace ThreadedSkinning
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()

			comboBoxEdit1.EditValue = DevExpress.LookAndFeel.UserLookAndFeel.Default.SkinName
			For Each cnt As DevExpress.Skins.SkinContainer In DevExpress.Skins.SkinManager.Default.Skins
				comboBoxEdit1.Properties.Items.Add(cnt.SkinName)
			Next cnt
		End Sub
		Private Sub comboBoxEdit1_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxEdit1.SelectedValueChanged
			DevExpress.LookAndFeel.UserLookAndFeel.Default.SkinName = comboBoxEdit1.Text
		End Sub

		Private Class ShowFormThreadData
			Public mainLookAndFeel As DevExpress.LookAndFeel.UserLookAndFeel
			' TODO add more fields to pass data to a new thread
		End Class

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Dim thread As New Thread(New ParameterizedThreadStart(AddressOf myThread_OpenForm))
			thread.IsBackground = True
			thread.SetApartmentState(ApartmentState.STA)
			Dim data As New ShowFormThreadData()
			data.mainLookAndFeel = DevExpress.LookAndFeel.UserLookAndFeel.Default
			thread.Start(data)
		End Sub

		Public Shared Sub myThread_OpenForm(ByVal data As Object)
			Dim td As ShowFormThreadData = TryCast(data, ShowFormThreadData)

			' this thread 
			DevExpress.UserSkins.OfficeSkins.Register()
			DevExpress.UserSkins.BonusSkins.Register()
			DevExpress.LookAndFeel.UserLookAndFeel.Default.Assign(td.mainLookAndFeel)

			Dim form2 As New Form2()
			AddHandler td.mainLookAndFeel.StyleChanged, AddressOf form2.MainLookAndFeel_StyleChanged
			form2.ShowDialog()
		End Sub
	End Class
End Namespace

File: CS/ThreadedSkinning/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Threading;
using DevExpress.LookAndFeel;
using DevExpress.XtraEditors;

namespace ThreadedSkinning {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();

            comboBoxEdit1.EditValue = DevExpress.LookAndFeel.UserLookAndFeel.Default.SkinName;
            foreach(DevExpress.Skins.SkinContainer cnt in DevExpress.Skins.SkinManager.Default.Skins) {
                comboBoxEdit1.Properties.Items.Add(cnt.SkinName);
            }
        }
        private void comboBoxEdit1_SelectedValueChanged(object sender, EventArgs e) {
            DevExpress.LookAndFeel.UserLookAndFeel.Default.SkinName = comboBoxEdit1.Text;
        }

        private class ShowFormThreadData {
            public DevExpress.LookAndFeel.UserLookAndFeel mainLookAndFeel;
            // TODO add more fields to pass data to a new thread
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            Thread thread = new Thread(new ParameterizedThreadStart(myThread_OpenForm));
            thread.IsBackground = true;
            thread.SetApartmentState(ApartmentState.STA);
            ShowFormThreadData data = new ShowFormThreadData();
            data.mainLookAndFeel = DevExpress.LookAndFeel.UserLookAndFeel.Default;
            thread.Start(data);
        }

        public static void myThread_OpenForm(object data) {
            ShowFormThreadData td = data as ShowFormThreadData;

            // this thread 
            DevExpress.UserSkins.OfficeSkins.Register();
            DevExpress.UserSkins.BonusSkins.Register();
            DevExpress.LookAndFeel.UserLookAndFeel.Default.Assign(td.mainLookAndFeel);
            
            Form2 form2 = new Form2();
            td.mainLookAndFeel.StyleChanged += new EventHandler(form2.MainLookAndFeel_StyleChanged);
            form2.ShowDialog();
        }
    }
}